src/java.base/share/classes/jdk/internal/foreign/NativeMemorySegmentImpl.java

```diff
@@ -52,9 +52,9 @@ public sealed class NativeMemorySegmentImpl extends AbstractMemorySegmentImpl pe
     final long min;
 
     @ForceInline
-    NativeMemorySegmentImpl(long min, long length, boolean readOnly, SegmentScope scope) {
+    NativeMemorySegmentImpl(long min, long length, boolean readOnly, MemorySessionImpl scope) {
         super(length, readOnly, scope);
-        this.min = (UNSAFE.addressSize() == 4)
+        this.min = (Unsafe.ADDRESS_SIZE == 4)
```

Comment:
Strangely, using the instance method here causes initialization failures when doing stuff like:
```
Arena.ofAuto().allocate(...)
```